sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java

```diff
@@ -95,7 +94,7 @@ public CoverageDetail apply(@Nonnull String testName) {
       builder.setTestName(testName);
       for (CoverageBlock block : testCase.coverageBlocks()) {
         coveredBuilder.clear();
-        coveredBuilder.setFileRef(resourceCache.get(block.testable().component().key()).batchId());
+        coveredBuilder.setFileRef(componentCache.get(((DefaultTestable) block.testable()).inputFile().key()).batchId());
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png)![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png)NullPointerException might be thrown as 'get' is nullable here[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)